src/audio/component.c

```diff
@@ -163,12 +163,6 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 	case COMP_TRIGGER_RESET:
 		/* reset always succeeds */
 		dev->state = COMP_STATE_READY;
-		if (dev->state == COMP_STATE_ACTIVE ||
-			dev->state == COMP_STATE_PAUSED) {
-			trace_comp_error("CER");
-			trace_error_value(dev->state);
-			ret = 0;
-		}
```

Comment:
maybe the intention from author is that we can't reset at running/paused(can do that on stopped only), something like this?

 	case COMP_TRIGGER_RESET:
               /* Keyon: can't reset at running/paused */
		if (dev->state == COMP_STATE_ACTIVE ||
			dev->state == COMP_STATE_PAUSED) {
			trace_comp_error("CER");
			trace_error_value(dev->state);
		}
 		/* reset always succeeds */
	 	dev->state = COMP_STATE_READY;
		ret = 0;
                break;

Reply:
that means only report a warning here but still let pipeline can be reset.

Reply:
I am not sure of this. what is the designer's target? @lgirdwood could you clarify this?

Reply:
normally when the reset command comes,  it must be accepted.

Reply:
reset must always succeed. Looking again it's the dev->state = COMP_STATE_READY; that needs removed in the reset case. The state condition check here are more for information purposes.

Reply:
so I suggested to draw a state machine diagram to make them clear, and fix those kinds of potential issues thoroughly.

Reply:
There is a diagram in the code.

Reply:
yes, in component.h, something might be outdated, @zhigang-wu can you please help update it.

-    /* audio component states
|     * the states may transform as below:
|     *        new()         params()          start()
|     * none -----> init ------> setup -----> running
|     * none <----- init <------ setup <----- running
|     *        free()          reset()             stop()
|     */

Reply:
ok. no problem.